src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2635,40 +2596,51 @@ internal static int IndexOfAnyValueType<T>(ref T searchSpace, T value0, T value1
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static int ComputeLastIndex<T>(nint offset, Vector128<T> equals) where T : struct
-        {
-            uint notEqualsElements = equals.ExtractMostSignificantBits();
-            int index = 31 - BitOperations.LeadingZeroCount(notEqualsElements); // 31 = 32 (bits in Int32) - 1 (indexing from zero)
-            return (int)offset + index;
-        }
-
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static int ComputeLastIndex<T>(nint offset, Vector256<T> equals) where T : struct
+        private static int ComputeLastIndex<TVector, T>(nint offset, TVector equals)
```

Comment:
This is working around the lack of variable sized `ExtractMostSignificantBits`.

It's not pretty, but it works and ideally gets replaced in .NET 8 as part of the `VectorMask` work...